src/main.cc

```diff
@@ -288,11 +288,212 @@ NAN_METHOD(EnumKeys) {
     RegCloseKey(hCurrentKey);
 }
 
-NAN_MODULE_INIT(Init) {
-  Nan::SetMethod(target, "readValues", ReadValues);
-  Nan::SetMethod(target, "enumKeys", EnumKeys);
-}
+  NAN_METHOD(CreateKey)
+  {
+    auto argCount = info.Length();
+    if (argCount != 1 && argCount != 2)
+    {
+      Nan::ThrowTypeError("Wrong number of arguments");
+      return;
+    }
+
+    if (!info[0]->IsNumber())
+    {
+      Nan::ThrowTypeError("A number was expected for the first argument, but wasn't received.");
+      return;
+    }
+
+    auto first = reinterpret_cast<HKEY>(Nan::To<int64_t>(info[0]).FromJust());
+
+    HKEY hCurrentKey = first;
+    if (argCount == 2 && !info[1]->IsNullOrUndefined())
+    {
+      if (!info[1]->IsString())
+      {
+        Nan::ThrowTypeError("A string was expected for the second argument, but wasn't received.");
+        return;
+      }
+      Nan::Utf8String subkeyArg(Nan::To<v8::String>(info[1]).ToLocalChecked());
+      auto keyPath = utf8ToWideChar(std::string(*subkeyArg));
+      if (keyPath == nullptr)
+      {
+        Nan::ThrowTypeError("A string was expected for the second argument, but could not be parsed.");
+        return;
+      }
+      auto openKey = RegCreateKeyEx(
+          first,
+          keyPath,
+          0,
+          nullptr,
+          REG_OPTION_NON_VOLATILE,
+          KEY_READ | KEY_WOW64_64KEY,
+          nullptr,
+          &hCurrentKey,
+          nullptr);
+      if (openKey != ERROR_SUCCESS)
+      {
+        // FIXME: the key does not exist, just return an empty array for now
+        info.GetReturnValue().Set(New<v8::Boolean>(false));
+        return;
+      }
+    }
+
+    info.GetReturnValue().Set(New<v8::Boolean>(true));
+    if (hCurrentKey != first)
+      RegCloseKey(hCurrentKey);
+  }
+
+  NAN_METHOD(SetValue)
+  {
+    {
+      auto argCount = info.Length();
+      if (argCount < 5)
+      {
+        Nan::ThrowTypeError("Wrong number of arguments");
+        return;
+      }
+
+      if (!info[0]->IsNumber())
+      {
+        Nan::ThrowTypeError("A number was expected for the first argument, but wasn't received.");
+        return;
+      }
+
+      if (!info[1]->IsString())
+      {
+        Nan::ThrowTypeError("A string was expected for the second argument, but wasn't received.");
+        return;
+      }
+
+      if (!info[2]->IsString())
+      {
+        Nan::ThrowTypeError("A number was expected for the third argument, but wasn't received.");
+        return;
+      }
+
+      if (!info[3]->IsString())
+      {
+        Nan::ThrowTypeError("A number was expected for the fourth argument, but wasn't received.");
+        return;
+      }
+
+      auto first = reinterpret_cast<HKEY>(Nan::To<int64_t>(info[0]).FromJust());
+
+      HKEY hCurrentKey = first;
+      if (argCount == 5 && !info[1]->IsNullOrUndefined() && !info[2]->IsNullOrUndefined() && !info[3]->IsNullOrUndefined())
+      {
```

Comment:
Instead of having this huge `if` body, you can invert the condition and turn it into a early return like the ones above. Or just use the `if`s above to also check for `IsNullOrUndefined` and return early.